### PR TITLE
Consistent time handling and generic time converter code

### LINKED ISSUE #2748: Incosistent time handling in agenda import/export
OS offers an example CSV file to see how agenda entries are added to a CSV import file. This file uses the format "hh:mm" for the duration. So let's test it:

- download the offered CSV example file
- use this file for an import
- now export the resulting agenda

result: in the exported CSV the duration has the format "mm". This is inconsistent and confusing although the exported file can be used for another import.

To be consistent either the format "hh:mm" or "mm" should be used for import and export.

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Closes #2748 and removed some more code repetitions. Also this factory could be extended if more parse options are needed.

### CODE REVIEW

File: openslides/core/static/js/core/base.js [hunk 1]
```diff
@@ -852,48 +852,94 @@ angular.module('OpenSlidesApp.core', [
     }
 ])
 
+/* Two functions to convert between time duration in seconds <-> human readable time span.
+ * E.g. 90 sec <-> 1:30 (min), 3661 sec <-> 1:01:01 (h)
+ *
+ * secondsToHumanTime: Expects seconds and give [h*:]mm[:ss]. The minutes part is always given, the hours
+ *      and minutes could be controlled. The default are forced seconds and hours just if it is not 0.
+ *      - forceSeconds (default true): show seconds even if 00
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: If I understand this correctly we have 3 cases. Always seconds, only seconds if they are greater than 0, no seconds.

Am I right? If yes please implement this using only one option flag (accepting strings like 'enable', 'auto', 'disable' or something like this) and not two bool flag.

File: openslides/core/static/js/core/base.js [hunk 2]
```diff
@@ -852,48 +852,94 @@ angular.module('OpenSlidesApp.core', [
...
+ *      - ignoreSeconds (default false): Do not show seconds, even is greater 0. This overrides forceSeconds.
+ *      - forceHours: (default false): Show hours even is hours=0. Just one zero is appended.
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Here the same.

File: openslides/core/static/js/core/base.js [hunk 3]
```diff
@@ -852,48 +852,94 @@ angular.module('OpenSlidesApp.core', [
...
+                    time = h + ':' + time;
+                }
+                if (seconds < 0) {
+                    time = '-'+time;
+                }
+                return time;
+            },
+            humanTimeToSeconds: function (data, params) {
+                if (!params)
+                    params = {seconds: false, hours: false};
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Missing `{ }`

File: openslides/core/static/js/core/base.js [hunk 4]
```diff
@@ -852,48 +852,94 @@ angular.module('OpenSlidesApp.core', [
...
+                var minLength = 1;
+                if (params.seconds) minLength++;
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Missing `{ }`

File: openslides/core/static/js/core/base.js [hunk 5]
```diff
@@ -852,48 +852,94 @@ angular.module('OpenSlidesApp.core', [
...
+                if (params.hours) minLength++;
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Missing `{ }`

File: openslides/core/static/js/core/base.js (hunk 1 above)
Comment (FinnStutzenstein) [inline openslides/core/static/js/core/base.js]: Yes, good suggestion..

**Comment by FinnStutzenstein** [comment]
@normanjaeckel ok, all done and rebased.

File: openslides/agenda/static/js/agenda/csv.js [hunk 6]
```diff
@@ -5,16 +5,20 @@
...
             ];
             _.forEach(agenda, function (item) {
                 var row = [];
+                var duration = item.duration ? HumanTimeConverter.secondsToHumanTime(item.duration*60,
+                        { seconds: 'disabled',
+                            hours: 'enabled' }) : '';
                 row.push('"' + (item.title || '') + '"');
                 row.push('"' + (item.text || '') + '"');
-                row.push('"' + (item.duration || '') + '"');
+                row.push('"' + (duration || '') + '"');
```
Comment (normanjaeckel) [inline openslides/agenda/static/js/agenda/csv.js]: duration is already an empty string (if item.duration was falsy) so we do not have to use the or clause again.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Done.

**Pull request merged**